tests/unit/r2i/engine.cc

```diff
@@ -0,0 +1,96 @@
+/* Copyright (C) 2020 RidgeRun, LLC (http://www.ridgerun.com)
+ * All Rights Reserved.
+ *
+ * The contents of this software are proprietary and confidential to RidgeRun,
+ * LLC.  No part of this program may be photocopied, reproduced or translated
+ * into another programming language without prior written consent of
+ * RidgeRun, LLC.  The user is free to modify the source code after obtaining
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+*/
+
+#include <r2i/engine.h>
+
+#include <CppUTest/CommandLineTestRunner.h>
+#include <CppUTest/MemoryLeakDetectorNewMacros.h>
+#include <CppUTest/MemoryLeakDetectorMallocMacros.h>
+#include <CppUTest/TestHarness.h>
+
+namespace mock {
+class Preprocessing : public r2i::IPreprocessing {
+ public:
+  Preprocessing () {}
+  r2i::RuntimeError apply(r2i::IFrame &data) {
+    return r2i::RuntimeError();
+  }
+  std::vector<r2i::ImageFormat> getAvailableFormats() {
+    std::vector<r2i::ImageFormat> image_format;
+    return image_format;
+  }
+  std::vector<std::vector<int>> getAvailableDataSizes() {
+    std::vector<std::vector<int>> data_sizes;
```

Comment:
I know this is from a past PR but this is not going to be enough for our needs. How do you describe ranges?